scintillometry/utils/pint_utils.py

```diff
@@ -0,0 +1,169 @@
+           time objects' method.
+    """
+    def __init__(self, par_file=None, tim_file=None, ephem='de436',
+                 include_bipm=True, bipm_version='BIPM2015', include_gps=True,
+                 planets=False, tdb_method="default", **kwargs):
+        # Load model file
+        self.par_file = par_file
+        self.control_params = {'ephem': ephem, 'bipm_version': bipm_version,
+                               'include_bipm': include_bipm,
+                               'bipm_version': bipm_version,
+                               'include_gps': include_gps, 'planets': planets,
+                               'tdb_method': tdb_method}
+        self.control_params.update(kwargs)
+        # Load toa file
+        self.tim_file = tim_file
+
+    @property
+    def par_file(self):
+        return self._par_file
+
+    @par_file.setter
+    def par_file(self, val):
+        self._par_file = val
+        if val is not None:
+            self.model = model.get_model(val)
+        else:
+            self.model = None
+
+    @property
+    def tim_file(self):
+        return self._tim_file
+
+    @tim_file.setter
+    def tim_file(self, val):
+        self._tim_file = val
+        if val is not None:
+            self.get_toas(tim_file=val, **self.control_params)
+        else:
+            self.toas = val
+
+    def get_toas(self, tim_file=None, times=None, obs=None, obs_freq=None,
+                 **control_params):
+        """Read TOAs from either a .tim file or from a list the of timestamps.
+
+           Parameter
+           ---------
+           tim_file : str, optional.
+               File name of input .tim file.
+           times : `~astropy.time.Time` or list of `~astropy.time.Time`,
+               optional.
+               Input timestamps. Default is None. If 'times' is provided, both
+               'obs' and 'obs_freq' have to be provided.
+           obs : str, optinal
+               Observatory name or code, Default is None
+           obs_freq : float or `~astropy.units.Quantity`,
+               Observering frequency, Default is None
+           control_params : dict
+               The control parameters for constructing the TOA table. It
+               includes the solar system ephemeris (ephem), clock correction
+               files (gps, bipm).
+               See more detials at: https://github.com/nanograv/PINT
+
+           Note
+           ----
+           This class accepts two type of input for time, the Tempo/Tempo2 style
+           .tim and the `~astropy.time.Time` or a list of it. User can choose
+           either way of input, but have to choose one.
+
+           Raises
+           ------
+           RunTimeError
+               If the timestamps are not provided.
+           ValueError
+               If the observatory name or observing frequency are not provided,
+               when using timestamps.
+        """
+        # This enables user input control parameters from the this method.
```

Comment:
This comment is not very useful - why is it necessary to override? I guess in part I'm not sure why there is a separate `PintUtils` and `PintPhase` class. I think `PintUtils` is meant to be the equivalent of the my `Predictor` class?

Reply:
The `PintUtils` are designed to provide some convenient functions for basic PINT usages (e.g., compute the apparent spin frequency, compute the time delays, etc). If in the scintillometry analysis, we only care about the pulse phase, then the PintUtils are not necessary.